tests/unit/test_session.py

```diff
@@ -367,6 +367,15 @@ def test_credential_provider_not_called_when_creds_provided(self):
                          "explicit credentials were provided to the "
                          "create_client call.")
 
+class TestPerformOperation(BaseSessionTest):
+    def test_s3(self):
```

Comment:
Really? Interesting. In that case, I would just add an assertion of the response at the end to ensure the mocked response is properly returned by the ``call()``. I guess we were just relying on integration test for that.